hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/BrokenStoreFileCleaner.java

```diff
@@ -147,6 +147,10 @@ private void cleanFileIfNeeded(FileStatus file, HStore store, AtomicLong deleted
       return;
     }
 
+    //This last check ia needed in case the region got moved after we fetched the list of online
```

Comment:
A style problem?

And also better mention this

"Adding the check here is enough to prevent accidentally deleting of intact HFiles. If the region is still available, i.e, not closed or closing, all above checks are valid so we are safe to delete the 'broken' files. And if the region is closed or closing when we reach here, since we will always give up deleting, the only possible inconsistency is that the file is broken since the region is not closing or close when performing above checks but here we give up deleting it. So in either way, we will not accidentally deleting intact HFiles, but only give up deleting broken files. But it is OK since we can still delete them in the next round of cleaner run."